test/index.js

```diff
@@ -23,24 +24,23 @@ const {withUnstableArraySort} = require ('./quicksort');
 const {Nil, Cons} = List;
 const {Nothing, Just} = Maybe;
 
+//  mapArb :: (a -> b) -> Arbitrary a -> Arbitrary b
+const mapArb = f => arb => jsc.bless ({generator: arb.generator.map (f)});
+
+//  smapArb :: (a -> b) -> (b -> a) -> Arbitrary a -> Arbitrary b
+const smapArb = f => g => arb => arb.smap (f, g, show);
+
 //  Lazy$of :: a -> Lazy a
 function Lazy$of(x) {
   eq (arguments.length, Lazy$of.length);
   return Z.of (Lazy, x);
 }
 
 //  ListArb :: Arbitrary a -> Arbitrary (List a)
-const ListArb = arb => (
-  jsc.array (arb)
-  .smap (xs => {
-    let list = Nil;
-    for (let idx = 0; idx < xs.length; idx += 1) list = Cons (xs[idx], list);
-    return list;
-  })
-);
+const ListArb = arb => smapArb (arrayToList) (listToArray) (jsc.array (arb));
 
 //  MaybeArb :: Arbitrary a -> Arbitrary (Maybe a)
-const MaybeArb = arb => jsc.oneof (jsc.constant (Nothing), arb.smap (Just));
+const MaybeArb = arb => jsc.oneof (jsc.constant (Nothing), mapArb (Just) (arb));
```

Comment:
It might be worth adding a custom shrink function here to get back to an `Arb a` given an `Arb (Maybe a)`. :thinking:

Reply:
Hm. No. :P